### PR TITLE
V3.6 remote log

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: https://github.com/cocos/3d-tasks/issues/12598

### Changelog

* transport log to server with UDP packages
* reduce code size by shorten log format

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/base/Log.cpp [hunk 1]
```diff
@@ -46,11 +60,229 @@
...
 
 #elif (CC_PLATFORM == CC_PLATFORM_ANDROID)
+
     #include <android/log.h>
+
 #elif CC_PLATFORM == CC_PLATFORM_OHOS
     #include <hilog/log.h>
 #endif
 
+#if CC_REMOTE_LOG
```
Comment (minggo) [inline native/cocos/base/Log.cpp]: Can move these codes into another file?

Comment (PatriceJiang) [reply native/cocos/base/Log.cpp]: yes

File: native/cocos/base/Log.cpp [hunk 2]
```diff
@@ -197,6 +207,13 @@ void Log::logMessage(LogType type, LogLevel level, const char *formats, ...) {
...
     fputs(buff, stdout);
 #endif
+
+    logRemote(buff);
 }
 
+#if !CC_REMOTE_LOG
+void Log::logRemote(const char *msg) {
+}
+#endif
```
Comment (minggo) [inline native/cocos/base/Log.cpp:212]: I think it is better to put these logic into LogRemote.cpp

File: native/cocos/base/Log.h [hunk 3]
```diff
@@ -54,6 +54,7 @@ class CC_DLL Log {
     static void setLogFile(const ccstd::string &filename);
     static void close();
     static void logMessage(LogType type, LogLevel level, const char *formats, ...);
+    static void logRemote(const char *msg);
```
Comment (minggo) [inline native/cocos/base/Log.h]: Not a public function.

File: native/cocos/base/LogRemote.cpp [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+    INITIALIZED,
+    CONFIGURED,
+    OK,
+    DONE, // FAILED
+};
+/**
+* Parse auto-test-config.json to get ServerConfig.IP & ServerConfig.PORT
+* Logs will be formated with 5 fields
+* 1. testId
+* 2. clentId
```
Comment (dumganhar) [inline native/cocos/base/LogRemote.cpp]: Typo: clientId?

File: native/cocos/base/LogRemote.cpp [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+* 3. bootId,  the boot timestamp
+* 4. millseconds since boot
```
Comment (dumganhar) [inline native/cocos/base/LogRemote.cpp]: Typo: milliseconds

File: native/cocos/base/LogRemote.cpp [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+
+        {
+            // parse ip & port
+            rapidjson::Value &cfg = doc[ATC_KEY_CONFIG];
+            if (!cfg.HasMember(ATC_KEY_IP) || !cfg.HasMember(ATC_KEY_PORT)) {
+                _status = UdpLogClientState::DONE;
+                return;
+            }
+            const char *remoteIp = cfg[ATC_KEY_IP].GetString();
+            int remotePort = cfg[ATC_KEY_PORT].GetInt() + 1;
```
Comment (dumganhar) [inline native/cocos/base/LogRemote.cpp:186]: Add comment for the logic of `+1`.

File: native/cocos/base/LogRemote.cpp [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+            _status = UdpLogClientState::DONE;
+            return;
+        }
+
+    #if CC_PLATFORM == CC_PLATFORM_WINDOWS
+        u_long mode = 1;
+        if (ioctlsocket(_sock, FIONBIO, &mode)) {
+            auto errorCode = getErrorCode();
+            printf("set nonblock failed, code: %d\n", errorCode);
+            // continue
```
Comment (dumganhar) [inline native/cocos/base/LogRemote.cpp:247]: If ioctlsocket fails, this function still set `_status` to UdpLogClientState::OK , and run `send(_sock, msg.data(), msg.size(), 0);` logic ?

File: native/cocos/base/Log.cpp [hunk 8]
```diff
@@ -197,6 +207,8 @@ void Log::logMessage(LogType type, LogLevel level, const char *formats, ...) {
 #else
     fputs(buff, stdout);
 #endif
+
+    logRemote(buff);
```
Comment (dumganhar) [inline native/cocos/base/Log.cpp:211]:
```suggestion
    logRemote(buff);
```
```suggestion
#if CC_REMOTE_LOG
    logRemote(buff);
#endif
```

File: native/cocos/bindings/jswrapper/config.cpp [hunk 9]
```diff
@@ -23,72 +23,15 @@
...
-    delete[] buf;
+#include "bindings/jswrapper/config.h"
+#include <cstdarg>
+#include "base/Log.h"
+
+void selogMessage(cc::LogLevel level, const char *tag, const char *format, ...) {
+    char logbuf[512] = {0};
+    va_list argp;
+    va_start(argp, format);
+    (void)std::vsnprintf(logbuf, sizeof(logbuf), format, argp);
```
Comment (dumganhar) [inline native/cocos/bindings/jswrapper/config.cpp:34]: Why add `(void)` prefix before std::vsnprintf ?

File: native/cocos/bindings/manual/jsb_platform_android.cpp [hunk 10]
```diff
@@ -54,26 +52,26 @@ static bool JSB_loadFont(se::State &s) {
 
         ccstd::string originalFamilyName;
         ok &= sevalue_to_native(args[0], &originalFamilyName);
-        SE_PRECONDITION2(ok, false, "JSB_loadFont : Error processing argument: originalFamilyName");
+        SE_PRECONDITION2(ok, false, "originalFamilyName");
```
Comment (dumganhar) [inline native/cocos/bindings/manual/jsb_platform_android.cpp]:
```suggestion
        SE_PRECONDITION2(ok, false, "originalFamilyName");
```
```suggestion
        SE_PRECONDITION2(ok, false, "Error processing arguments");
```

File: native/cocos/bindings/manual/jsb_platform_android.cpp [hunk 11]
```diff
@@ -54,26 +52,26 @@ static bool JSB_loadFont(se::State &s) {
...
 
         ccstd::string source;
         ok &= sevalue_to_native(args[1], &source);
-        SE_PRECONDITION2(ok, false, "JSB_loadFont : Error processing argument: source");
+        SE_PRECONDITION2(ok, false, "source");
```
Comment (dumganhar) [inline native/cocos/bindings/manual/jsb_platform_android.cpp]: Error processing arguments

File: native/cocos/bindings/manual/jsb_platform_apple.mm [hunk 12]
```diff
@@ -98,7 +98,7 @@ static bool JSB_loadFont(se::State &s) {
 
         ccstd::string originalFamilyName;
         ok &= sevalue_to_native(args[0], &originalFamilyName);
-        SE_PRECONDITION2(ok, false, "JSB_loadFont : Error processing argument: originalFamilyName");
+        SE_PRECONDITION2(ok, false, "originalFamilyName");
```
Comment (dumganhar) [inline native/cocos/bindings/manual/jsb_platform_apple.mm]: Error processing arguments

File: native/cocos/bindings/manual/jsb_platform_linux.cpp [hunk 13]
```diff
@@ -49,11 +49,11 @@ static bool JSB_loadFont(se::State &s) {
 
         ccstd::string originalFamilyName;
         ok &= sevalue_to_native(args[0], &originalFamilyName);
-        SE_PRECONDITION2(ok, false, "JSB_loadFont : Error processing argument: originalFamilyName");
+        SE_PRECONDITION2(ok, false, "originalFamilyName");
 
         ccstd::string source;
         ok &= sevalue_to_native(args[1], &source);
-        SE_PRECONDITION2(ok, false, "JSB_loadFont : Error processing argument: source");
+        SE_PRECONDITION2(ok, false, "source");
```
Comment (dumganhar) [inline native/cocos/bindings/manual/jsb_platform_linux.cpp]: Error processing arguments

File: native/cocos/base/LogRemote.cpp (hunk 7 above)
Comment (PatriceJiang) [reply native/cocos/base/LogRemote.cpp:247]: Setting the fd to nonblock is not required

File: native/cocos/bindings/jswrapper/config.cpp (hunk 9 above)
Comment (PatriceJiang) [reply native/cocos/bindings/jswrapper/config.cpp:34]: To ignore return value

**Review by minggo — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Pull request merged**